client/pages/quiz/[slug].js

```diff
@@ -0,0 +1,166 @@
+/*
+  This page will load at the url "/quiz/:slug"
+*/
+
+import React, { useState, useEffect } from "react";
+// import useSWR from 'swr';// <-- uncomment when DB is ready
+import { useRouter } from "next/router";
+// import apiRoutes from '../../api-routes';// <-- uncomment when DB is ready
+import { dummyData } from "../dummy-quiz-data";
+// import DisplayMessage from '../../components/shared/DisplayMessage';
+import {
+  QuizHeader,
+  AnswersTileSection,
+  NextQuestionBtnContainer,
+  AnswerTileContainerLink,
+} from "../../components/quizSingle/styles";
+import { PageHeader } from "../../components/shared/styles";
+import Header from "../../components/header/Header";
+import QuestionHeader from "../../components/quizSingle/QuestionHeader";
+import AnswerTileContainer from "../../components/quizSingle/AnswerTileContainer";
+import NextQuestionBtn from "../../components/quizSingle/NextQuestionBtn";
+import { ContentWrapper } from "../../components/footer/styles";
+
+export default function Quiz() {
+  const router = useRouter();
+
+  // const fetcher = url => fetch(url).then(res => res.json());
+  // const {data, error} = useSWR(apiRoutes.getAllQuizzes, fetcher);
+  const [allSubjectQuizzes, setAllSubjectQuizzes] = useState([]);
+  const [chosenQuiz, setChosenQuiz] = useState({});
+  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
+  const [allQuestionsCount, setAllQuestionsCount] = useState(0);
+  const [currentQuestion, setCurrentQuestion] = useState({});
+  const [currentQuestionAnswers, setCurrentQuestionAnswers] = useState([]);
+  const [selectedAnswers, setSelectedAnswers] = useState({});
+
+  // When id is selected and quizzes are loaded,
+  // find quiz by id and save it to state
+  useEffect(() => {
+    if (router.query.slug) {
```

Comment:
This logic will change soon as we have DB routes to get the quiz questions and answer so this confused me at first (querying all the quizzes to find the right one.) No worries here though, this works great for the dummy data and we can update it once the DB is live.